tools/pythonpkg/tests/fast/api/test_read_csv.py

```diff
@@ -469,10 +469,12 @@ def test_read_csv_combined(self, duckdb_cursor):
         assert rel.columns == rel2.columns
         assert rel.types == rel2.types
 
-    def test_read_csv_names(self):
+    def test_read_csv_names(self, tmp_path):
```

Comment:
I am not entirely confident this will be unique for every test

I have more faith in this fixture, do you mind using this?
```py
@pytest.fixture(scope="function")
def temp_file_name(request, tmp_path_factory):
    return str(tmp_path_factory.mktemp(request.function.__name__, numbered=True) / 'file.csv')
```

tmp_path_factory creates unique paths, and numbers them, + for good measure I included the name of the test as another unique token

Reply:
> I am not entirely confident this will be unique for every test

The [pytest documentation for `tmp_path`](https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html#the-tmp-path-fixture) says almost verbatim exactly that which is why I hose it :)

![image](https://github.com/duckdb/duckdb/assets/417981/1e00be27-3616-42f3-8346-af90de6c0453)

Reply:
The implementation of `tmp_path` is very similar to your proposed implementation (I removed the docstring and clean up code that follows the `yield`):

```
def _mk_tmp(request: FixtureRequest, factory: TempPathFactory) -> Path:
    name = request.node.name
    name = re.sub(r"[\W]", "_", name)
    MAXVAL = 30
    name = name[:MAXVAL]
    return factory.mktemp(name, numbered=True)


@fixture
def tmp_path(
    request: FixtureRequest, tmp_path_factory: TempPathFactory
) -> Generator[Path, None, None]:
    path = _mk_tmp(request, tmp_path_factory)
    yield path
```